test/api/teaching_period_test.rb

```diff
@@ -9,6 +9,14 @@ def app
     Rails.application
   end
 
+  def test_get_teaching_periods_details
+    tp = TeachingPeriod.second
+    id_of_tp = tp.id
+
+    get with_auth_token "/api/teaching_periods/#{tp.id}"
+    assert_equal 200, last_response.status
```

Comment:
We want to do more than test that it returned 200 - did it just return and give you nothing? Is that ok?

Reply:
Dear Andrew, 

As far as I know (unless I am wrong), we need to check whether the returned teaching period has the required details. I am wondering what the key details that I should check are and how can I find out...? 

I am looking at a quite similar test in auth_test.rb that also check for user's details and I think I can mock this test. Below are the lines of the method:

# Test POST for new authentication token
  def test_auth_post
    data_to_post = {
      username: 'acain',
      password: 'password'
    }
    # Get response back for logging in with username 'acain' password 'password'
    post_json '/api/auth.json', data_to_post
    actual_auth = last_response_body
    expected_auth = User.first

    # Check that response contains a user.
    assert actual_auth.key?('user'), 'Expect response to have a user'
    assert actual_auth.key?('auth_token'), 'Expect response to have a auth token'

    response_user_data = actual_auth['user']

    **# Check that the returned user has the required details.
    # These match the model object... so can compare in loops
    user_keys = %w(id email first_name last_name username nickname receive_task_notifications receive_portfolio_notifications receive_feedback_notifications opt_in_to_research has_run_first_time_setup)

    assert_json_matches_model(response_user_data, expected_auth, user_keys)

    user_keys.each { |k| assert response_user_data.key?(k), "Response has key #{k}" }
    user_keys.each { |k| assert_equal expected_auth[k], response_user_data[k], "Values for key #{k} match" }**

    # Check other values returned
    assert_equal expected_auth.name, response_user_data['name'], 'Names match'
    assert_equal expected_auth.role.name, response_user_data['system_role'], 'Roles match'

    assert_equal expected_auth.auth_token, actual_auth['auth_token']
  end

The bold lines are what I think a similar test, we should have 'detail keys' so that we can compare expected values with actual values returned from getting the 'teaching period', shouldn't we? If that's the case, what should be the keys here in teaching period? 

I could not search anywhere any tutorials teaching how to write ruby tests so I just have been going through all other available tests of other modules, trying to find similar tests that I can learn from.

Please kindly suggest whether I am thinking in the right direction.

Linh Tran